File: frontend/lib/models/lottery/lottery_ticket_model.dart
class LotteryTicketModel {
  final String id;
  final String userId;
  final String username;
  final int drawNumber;
  final List<int> numbers;
  final bool isQuickPick;
  final int multiplier; // <--- Multiplier added here
  final List<int> matchedNumbers;
  final int matchCount;
  final String result;
  final String? tier;
  final String? tierLabel;
  final int winAmount;
  final DateTime createdAt;

  LotteryTicketModel({
    required this.id,
    required this.userId,
    required this.username,
    required this.drawNumber,
    required this.numbers,
    this.isQuickPick = false,
    this.multiplier = 1, 
    this.matchedNumbers = const [],
    this.matchCount = 0,
    this.result = 'PENDING',
    this.tier,
    this.tierLabel,
    this.winAmount = 0,
    required this.createdAt,
  });

  bool get isPending => result == 'PENDING';
  bool get isWin => result == 'WIN';
  bool get isLose => result == 'LOSE';

  factory LotteryTicketModel.fromJson(Map<String, dynamic> json) {
    return LotteryTicketModel(
      id: json['_id'] ?? '',
      userId: json['userId'] ?? '',
      username: json['username'] ?? 'Anonymous',
      drawNumber: (json['drawNumber'] as num?)?.toInt() ?? 0,
      numbers: List<int>.from(
        (json['numbers'] ?? []).map((e) => (e as num).toInt()),
      ),
      isQuickPick: json['isQuickPick'] ?? false,
      multiplier: (json['multiplier'] as num?)?.toInt() ?? 1, // <--- Multiplier parsed here
      matchedNumbers: List<int>.from(
        (json['matchedNumbers'] ?? []).map((e) => (e as num).toInt()),
      ),
      matchCount: (json['matchCount'] as num?)?.toInt() ?? 0,
      result: json['result'] ?? 'PENDING',
      tier: json['tier'],
      tierLabel: json['tierLabel'],
      winAmount: (json['winAmount'] as num?)?.toInt() ?? 0,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'].toString()) ?? DateTime.now()
          : DateTime.now(),
    );
  }
}

File: backend/src/models/lottery/lotteryTicket.model.js
const mongoose = require("mongoose");

const lotteryTicketSchema = new mongoose.Schema({
  userId:         { type: String, required: true },
  username:       { type: String, default: "Anonymous" },
  drawNumber:     { type: Number, required: true },
  numbers:        { type: [Number], required: true },
  isQuickPick:    { type: Boolean, default: false },

  // ── CRITICAL: must be in schema or Mongoose silently strips it ──
  // executeDraw reads this to scale prize: basePrize × multiplier
  // Missing field = always reads as undefined → defaults to 1 → wrong prize
  multiplier:     { type: Number, enum: [1, 2, 5, 10], default: 1 },

  matchedNumbers: { type: [Number], default: [] },
  matchCount:     { type: Number, default: 0 },
  result:         { type: String, enum: ["PENDING", "WIN", "LOSE"], default: "PENDING" },
  tier:           { type: String, default: null },
  tierLabel:      { type: String, default: null },
  winAmount:      { type: Number, default: 0 },
  createdAt:      { type: Date, default: Date.now },
});

module.exports = mongoose.models.LotteryTicket
  || mongoose.model("LotteryTicket", lotteryTicketSchema);